functions/legacy.js

```diff
@@ -130,6 +130,31 @@ module.exports = {
             needsMutableContent = true;
           }
 
+          const addAttachment = (url, contentType) => {
+            if (!url) {
+              return;
+            }
+
+            if (!payload.apns.payload.attachment) {
+              payload.apns.payload.attachment = {};
+            }
+
+            if (!payload.apns.payload.attachment['content-type']) {
+              payload.apns.payload.attachment['content-type'] = contentType;
+            }
+
+            if (!payload.apns.payload.attachment.url) {
+              payload.apns.payload.attachment.url = url;
+            }       
+
+            needsCategory = true;
+            needsMutableContent = true;
+          };
+            
+          addAttachment(req.body.data.video, 'mpeg4');
+          addAttachment(req.body.data.image, 'jpeg');
```

Comment:
Will this break if someone tries to pass, say a PNG under `image`?

Reply:
No, this is just a hint. The difference between images and video/audio playback on iOS is drastically different, so it's mostly hinting to use the right subsystem, which itself doesn't care about type (it's just a generic "load data"). For example, if I rename a png to have a completely-wrong extension (like `mkv` or something) it still works with a content type of `jpeg`.